File: solutions/python/leap/3/leap.py
"""
A function to report if a given year is a leap year.
"""


def generate_leap_years(limit_year):
    return (
        year
        for year in range(1, limit_year + 1)
        if year % 4 == 0 and year % 100 != 0 or year % 400 == 0
    )


def leap_year(year):
    return year in generate_leap_years(year)

File: solutions/java/leap/3/src/main/java/Leap.java
@FunctionalInterface
interface LeapYearChecker { 
    boolean isLeapYear(int year); 
}

class Leap {

    LeapYearChecker checker;

    public Leap() {
        checker = (year) -> year % 4 == 0 && (year % 100 != 0 || year % 400 == 0) ;
    }

    public boolean isLeapYear(int year) {
        return checker.isLeapYear(year);
    }
    
}
